src/BloomExe/Edit/EditingView.cs

```diff
@@ -779,6 +782,29 @@ void SaveChangedImage(GeckoHtmlElement imageElement, PalasoImage imageInfo, stri
 			}
 		}
 
+		/// <summary>
+		/// Check whether the new image file is the same as the one we already have chosen.
+		/// </summary>
+		/// <remarks>
+		/// If the user goes out of his way to choose the exact same picture file from the
+		/// original location again, a copy will still be created with a slightly revised
+		/// name.  This could be detected by comparing the basename, the file size, the
+		/// creation date, and then the actual bytes in the file, but that's fairly expensive
+		/// for a relatively uncommon occurrence.  We could perhaps use a tool to remove
+		/// unused picture files from a book's folder.
+		/// </remarks>
+		private bool IsSameFile(GeckoHtmlElement imageElement, PalasoImage imageInfo)
+		{
+			var src = imageElement.GetAttribute("src");
+			if (!String.IsNullOrEmpty(src))
+			{
+				var path = Path.Combine(_model.CurrentBook.FolderPath, src.Replace("%20", " "));
```

Comment:
%20 is the obvious one, but presumably what we really need the reverse whatever function put it there. UrlDecode?